util/travis/before_install.sh

```diff
@@ -8,6 +8,8 @@ if [[ $TRAVIS_OS_NAME == "linux" ]]; then
 		sudo apt-get update
 		sudo apt-get install llvm-3.1
 		sudo apt-get install clang
+	elif [[ $CC == "gcc" ]] && [[ $PLATFORM == "UnixCpp11" ]]; then
+		sudo apt-get install -y g++-6 gcc-6
 	fi
 	sudo apt-get update
 	sudo apt-get install p7zip-full
```

Comment:
Instead:

* Change this to `sudo apt-get install p7zip-full $COMPILER`
* Remove the distinction between `Unix` and `UnixCpp11`
* Add `COMPILER=g++` to the normal targets
* Make the script.sh check for `$COMPILER == "g++-6"`